packages/liveblocks-client/src/room.ts

```diff
@@ -1228,15 +1226,11 @@ See v0.13 release notes for more information.
   let _getInitialStatePromise: Promise<void> | null = null;
   let _getInitialStateResolver: (() => void) | null = null;
 
-  function getStorage<TRoot extends LsonObject>(): Promise<{
-    root: LiveObject<TRoot>;
+  async function getStorage<TStorage extends LsonObject>(): Promise<{
```

Comment:
This is the result of TypeScript convert this function to an async function.

Reply:
I remember doing the exact opposite because regenerator runtime was causing an issue in specific contexts, but I don't remember why 🤔 

Do you remember @ofoucherot? Was it for some SSR or node issues? Or just to reduce bundle size?

https://github.com/liveblocks/liveblocks/pull/59/commits/8a0e092bad2e748bef6d73db9771908b2dd618e0

Reply:
Ohhh, I see where I made the mistake in my reasoning, I think. I must have momentarily gotten my JS history wrong, and mistakenly thought `async`/`await` were part of the `es2015`, but they aren't 🤦

The correct history, for reference (I just looked it up 😉):

- `es5` had nothing fancy
- `es2015` (aka `es6`) introduced the Promise API, and generator functions
- `es2017` (aka `es7`) introduced `async`/`await` keywords

Since we're targeting `es2015` for our CJS bundles, TypeScript should transpile away any use of `async`/`await` and indeed emit the regenerator runtime code in our bundle. That would definitely increase the bundle size for the CJS part!

I agree that using language patterns that needlessly pump the bundle size should be avoided if possible, so in this case, it makes perfect sence to use the Promise API directly and avoid the use of `async`/`await`, because then TypeScript can just emit the code without transpiling.

I think we can forbid usage of certain language constructs with a simple ESLint rule, and provide exactly this rationale there. For public libraries that makes perfect sence!

Reply:
Will close this PR in favor of one that adds an ESLint rule to forbid the use of `async`/`await` keywords, and provide this rationale.